### PR TITLE
Plugin Instruction pages

### PR DESCRIPTION
**Pull request by jr0d**

### CODE REVIEW

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+</div>
+
+<h2>Configure kubect to Use the Plugin</h2>
+<p>These commands will crate a new kubeconfig context that instruct kubectl to delegate authentication tasks to
+the Konvoy credentials plugin</p>
+
+<h3>Create a cluster configuration</h3>
+<div class="command">
+  <button class="btn" style="float:right" data-clipboard-snippet="">
+    <img class="clippy" width="13" src="{{ .webPathPrefix }}static/clippy.svg" alt=""/>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:63]: "clippy" ... hehe...

image

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    <p>This is the CA certificate needed to validate connections to your cluster. If your cluster is using a well known
+        certificate authority, such as Let's Encrypt, or you have added your CA certificate to your system certificate pool;
+        you may skip this section.</p>
+    <div class="command">
+
+        <button class="btn" style="float:right" data-clipboard-snippet="">
+            <img class="clippy" width="13" src="{{ .webPrefixPath }}static/clippy.svg" alt=""/>
+        </button>
+        <pre><code>mkdir -p ${HOME}/.kube/certs/{{ .clusterName }}/ &amp;&amp; cat &lt;&lt; EOF &gt; ${HOME}/.kube/certs/{{ .clusterName }}/k8s-ca.crt
+{{ .caPem }}
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:13]: Any reason we don't just inline the CA base64 encoded to save a step?

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+EOF</code></pre>
+    </div>
+{{ end }}
+
+<h2>Download and Install the Konvoy credentials plugin</h2>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:18]: Title Case or not title case.

```suggestion
<h2>Download And Install The Konvoy Credentials Plugin</h2>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+<p>The plugin is currently available for Linux and MacOSX. Use the following command appropriate
+  for your operating system< to download and install the plugin</p>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:20]:
```suggestion
  for your operating system to download and install the plugin</p>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    </code></pre>
+    {{ else }}
+        <pre><code>curl --create-dirs {{ .linuxURL }} \
+      -o ${HOME}/{{ .installPath}}} && \
+      chmod +x ${HOME}/{{ .installPath }}
+    </code></pre>
+    {{ end }}
+</div>
+
+<h2>Configure kubect to Use the Plugin</h2>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:56]: Questionable about the title casing of kubectl, but we should be consistent with whatever you chose above.

```suggestion
<h2>Configure kubectl To Use The Plugin</h2>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+<p>These commands will crate a new kubeconfig context that instruct kubectl to delegate authentication tasks to
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:57]:
```suggestion
<p>These commands will create a new kubeconfig context that instructs kubectl to delegate authentication tasks to
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      --certificate-authority=${HOME}/.kube/certs/{{ .clusterName }}/k8s-ca.crt \
+      --server={{ .kubeAPI }}
+{{- else }}
+  kubectl config set-cluster {{ .clusterName }} \
+      --server={{ .kubeAPI }}
+{{- end }}
+  </code></pre>
+</div>
+
+<h3>Create a kubeconfig User Profile</h3>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:77]:
```suggestion
<h3>Create A Kubeconfig User Profile</h3>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  </button>
+  <pre><code>kubectl config set-credentials {{ .profileName }} \
+      --exec-command={{ .runPath }} \
+      --exec-api-version=client.authentication.k8s.io/v1beta1 \
+      --exec-arg="-auth-url={{ .asyncAuthURL }}" \
+      --exec-arg="-kubeconfig-user={{ .profileName }}" \
+      --exec-arg="-ca-file=${HOME}/.kube/certs/{{ .clusterName }}/k8s-ca.crt"
+    </code></pre>
+</div>
+<h3>Create the context</h3>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:94]:
```suggestion
<h3>Create The Context</h3>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+        Organizing Cluster Access Using kubeconfig Files
+      </a>
+    </p><p>
+      <a href="https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/">
+        Configure Access to Multiple Clusters
+      </a>
+    </p>
+  </div>
+  <div id="plugin-config-form"></div>
+  <h2>Wrapping up</h2>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:50]:
```suggestion
  <h2>Wrapping Up</h2>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+  <p>At this point you can may begin using the kubectl. Try running a simple kubectl command: </p>
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:51]:
```suggestion
  <p>At this point you can may begin using kubectl. Try running a simple kubectl command: </p>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+  <div class="command">
+    <button class="btn" style="float:right" data-clipboard-snippet="">
+      <img class="clippy" width="13" src="{{ .Config.Web_Path_Prefix }}static/clippy.svg" alt=""/>
+    </button>
+    <pre><code>
+    kubectl get pods -A
+    </code></pre>
+    <p>
+      If this is the first time you've attempted authenticate using the plugin, a browser window should open
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:60]:
```suggestion
      If this is the first time you've attempted to authenticate using the plugin, a browser window should open
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+      to the Konvoy authentication page. After successfully authentication, you should see a pod listing in your
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:61]:
```suggestion
      to the Konvoy authentication page. After successfull authentication, you should see a pod listing in your
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+      terminal. The plugin will store your identity token for subsequent requests. Once your identity token
+      expires (24 hours by default) the plugin will once again direct you to the Konvoy authentication page where
+      you may re-authenticate.
+    </p>
+  </div>
+</div>
+<script>
+    // var instructionEventName = "instruction-data-ready";
+    // var eventDataReady = new Event(instructionEventName);
+    // document.addEventListener(instructionEventName, renderInstructionsHandler, false);
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:71]: please remove commented code

```suggestion
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+            .map(k => esc(k) + '=' + esc(params[k]))
+            .join('&');
+        console.log("QUERY: " + query + " " + typeof query);
+        if (query !== "") {
+            url += "?" + query
+        }
+        R.open('GET', url);
+        R.onload = function () {
+            let instructionsData = R.responseText;
+            // document.dispatchEvent(eventDataReady);
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:96]:
```suggestion
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+            renderInstructionsHandler(instructionsData);
+        };
+        R.send();
+    }
+
+    function renderInstructionsHandler(data) {
+        // let content = document.getElementById("plugin-config-form");
+        // content.insertAdjacentHTML("beforeend", instructionsData);
```
Comment (joejulian) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:104]:
```suggestion
```

**Comment by jr0d** [comment]
@joejulian thanks for the review.

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 16]
```diff
@@ -0,0 +1,112 @@
+<h1>Build kubeconfig with kubectl</h1>
+{{ if .caPem }}
+    <h2>Copy Cluster CA Certificate</h2>
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:3]:
```suggestion
    <h2>Copy cluster CA certificate</h2>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    <p>This is the CA certificate needed to validate connections to your cluster. If your cluster is using a well known
+        certificate authority, such as Let's Encrypt, or you have added your CA certificate to your system certificate pool;
+        you may skip this section.</p>
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:6]:
```suggestion
        you can skip this section.</p>
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+      --server={{ .kubeAPI }}
+{{- else }}
+  kubectl config set-cluster {{ .clusterName }} \
+      --server={{ .kubeAPI }}
+{{- end }}
+  </code></pre>
+</div>
+
+<h3>Create a kubeconfig User Profile</h3>
+<p>If you plan on authenticating to the same cluster with multiple users, you should created multiple user profiles</p>
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:78]:
```suggestion
<p>If you are authenticating to the same cluster, with multiple users, you should created multiple user profiles</p>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link href="{{ .Config.Web_Path_Prefix }}static/main.css" rel="stylesheet" type="text/css">
+  <link href="{{ .Config.Web_Path_Prefix }}static/styles.css" rel="stylesheet" type="text/css">
+  <link href="{{ .Config.Web_Path_Prefix }}static/tabs.css" rel="stylesheet" type="text/css">
+  <link rel="icon" href="{{ .Config.Web_Path_Prefix }}static/favicon.png">
+</head>
+
+<body class="theme-body">
+<div class="plugin-kubeconfig-container">
+  <h1>Konvoy kubectl Credentials Plugin</h1>
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:19]:
```suggestion
  <h1>Konvoy kubectl credentials plugin</h1>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+  <p>This document describes the process of configuring kubectl to use the Konvoy Credentials Plugin.
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:20]:
```suggestion
  <p>This document describes the process of configuring kubectl to use the Konvoy credentials plugin.
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 21]
```diff
@@ -0,0 +1,175 @@
...
+    This credentials plugin makes it easy to use external identity provider accounts with the kubernetes API.
+    When enabled the plugin will initiate authentication sessions and store identity tokens automatically.</p>
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:22]:
```suggestion
    When enabled the plugin initiates authentication sessions and stores identity tokens automatically.</p>
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+  <div id="plugin-select-cluster" style="display: none">
+    <p><label for="plugin-select-provider-id">Select Provider:</label>
+      <select id="plugin-select-provider-id" onchange="providerOnChange()"></select></p>
+
+    <p><label for="plugin-select-cluster-id">Select Cluster:</label>
+      <select id="plugin-select-cluster-id" onchange="clusterOnChange()"></select></p>
+  </div>
+  <div id="plugin-download-kubeconfig">
+    <h2>Generate a kubeconfig</h2>
+    <p>Follow the instructions below to configure a kubectl context for the currently selected cluster.
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:32]:
```suggestion
    <p>The following instructions describe how to configure a kubectl context for the currently selected cluster.
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 23]
```diff
@@ -0,0 +1,175 @@
...
+      You may also generate a kubeconfig for all clusters by downloading this
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:33]:
```suggestion
      You can also generate a kubeconfig for all clusters by downloading this
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+      <a href="{{ .Config.Web_Path_Prefix}}/plugin/kubeconfig">kubeconfig
+      </a>.
+    </p>
+<p>For more information on using and managing kubeconfig files, take a look at this documentation:
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:37]:
```suggestion
<p>For more information on using and managing kubeconfig files, refer to this documentation:
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 25]
```diff
@@ -0,0 +1,175 @@
...
+    <button class="btn" style="float:right" data-clipboard-snippet="">
+      <img class="clippy" width="13" src="{{ .Config.Web_Path_Prefix }}static/clippy.svg" alt=""/>
+    </button>
+    <pre><code>
+    kubectl get pods -A
+    </code></pre>
+    <p>
+      If this is the first time you've attempted authenticate using the plugin, a browser window should open
+      to the Konvoy authentication page. After successfully authentication, you should see a pod listing in your
+      terminal. The plugin will store your identity token for subsequent requests. Once your identity token
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:62]:
```suggestion
      terminal. The plugin stores your identity token for subsequent requests. After your identity token
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+      expires (24 hours by default) the plugin will once again direct you to the Konvoy authentication page where
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:63]:
```suggestion
      expires, by default in 24 hours,  the plugin again directs you to the Konvoy authentication page where
```

File: staging/dex-k8s-authenticator/html-templates/plugin.html [hunk 27]
```diff
@@ -0,0 +1,175 @@
...
+      you may re-authenticate.
```
Comment (russurquhart1) [inline staging/dex-k8s-authenticator/html-templates/plugin.html:64]:
```suggestion
      you can re-authenticate.
```

File: staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html (hunk 2 above)
Comment (jr0d) [reply staging/dex-k8s-authenticator/html-templates/kubectl-plugin-instructions.html:13]: Now that I have this: https://github.com/mesosphere/konvoy-async-auth/pull/12/files, :+1:

**Comment by jr0d** [comment]
@russurquhart1 thank you for the review!

**Pull request merged**